File: Test/Test/ViewController.m
//
//  ViewController.m
//  Test
//
//  Created by Yan Jin on 3/3/15.
//  Copyright (c) 2015 Yan Jin. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    NSString *str = [NSString stringWithFormat:@"hello"];
    printf("%s", [[self reverseString:str] UTF8String]);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (NSString *)reverseString:(NSString *)string {
    NSUInteger len = string.length;
    unichar *data = malloc(sizeof(unichar) * len);
    for (int i = 0; i < len; i++) {
        data[i] = [string characterAtIndex:len - 1 - i];
    }
    NSString *reversedString = [NSString stringWithCharacters:data length:len];
    return reversedString;
}

@end

File: 05 Web/05 Web/ViewController.swift
//
//  ViewController.swift
//  05 Web
//
//  Created by Yan Jin on 2/28/15.
//  Copyright (c) 2015 Yan Jin. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var showWeb: UIWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let url = NSURL(string: "http://m.facebook.com")
        let task = NSURLSession.sharedSession().dataTaskWithURL(url!) {
            (data, response, error) in
            var urlContent = NSString(data: data, encoding: NSUTF8StringEncoding)
            dispatch_async(dispatch_get_main_queue(), { () -> Void in
                self.showWeb.loadHTMLString(urlContent, baseURL:nil)
            })
        }
        task.resume()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

